scripts/simplify_network.py

```diff
@@ -738,6 +741,118 @@ def drop_isolated_nodes(n, threshold):
     return n
 
 
+def merge_into_network(n, aggregation_strategies=dict()):
+    """
+    Find isolated nodes in the network and merge those of them which have load
+    value below than a specified threshold into a single isolated node which
+    represents all the remote generation.
+
+    Parameters
+    ----------
+    n : PyPSA.Network
+        Original network
+    threshold : float
+        Load power used as a threshold to merge isolated nodes
+
+    Returns
+    -------
+    modified network
+    """
+    # keep original values of the overall load and generation in the network
+    # to track changes due to drop of buses
+    generators_mean_origin = n.generators.p_nom.mean()
+    load_mean_origin = n.loads_t.p_set.mean().mean()
+
+    n.determine_network_topology()
+
+    # duplicated sub-networks mean that there is at least one interconnection between buses
+    i_islands = n.buses[
+        (~n.buses.duplicated(subset=["sub_network"], keep=False))
+        & (n.buses.carrier == "AC")
+    ].index
```

Comment:
Does this account identifies isolated networks? This seems to me that nodes that are interconnected by a line but disconnected from the entire network are not well identified or am I wrong?

We could extend the working principle to "isolated networks" that may be defined as subnetworks that do not represent the larger national power system whose demand is low (below threshold).

To keep the approach simple we can also skip that

Reply:
Added a function to pick isolated networks using a number of the nodes as a criterium in the first approximation. Absolutely agree that it is possible to use other heuristics, the current implementation just generalises the merging approach for n_isolated_buses > 1.